### ISSUE TITLE
Cleanup hangs in operators-helm tutorial

### ISSUE THREAD

**Issue by niaomingjian** [operator-framework/operator-sdk#4383]
#### What did you do?

Running the helm-based operator sample in [tutorial](https://sdk.operatorframework.io/docs/building-operators/helm/tutorial)
Cleanup hangs at this step [cleanup](https://sdk.operatorframework.io/docs/building-operators/helm/tutorial/#cleanup).

#### What did you expect to see?
CR was deleted normally.

#### What did you see instead? Under which circumstances?
`kubectl delete -f config/samples/example_v1alpha1_nginx.yaml` couldn't return to prompt normally and CR was not deleted completely. Still saw it from Lens.

#### Environment
Windows 10 docker kubernetes

**Kubernetes cluster type:**
v1.15.5

`$ operator-sdk version`
helm-operator-v1.2.0
quay.io/operator-framework/operator-sdk:v1.2.0

`$ kubectl version`

Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

#### Additional context

````
metadata:
  creationTimestamp: '2021-01-11T04:01:28Z'
  finalizers:
    - uninstall-helm-release
````

I had to remove uninstall-helm-release finalizer to make the cleanup continue.  Refer to [1](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers), [2](https://github.com/kubernetes/kubernetes/issues/60538)
What is the **uninstall-helm-release** finalizer? Why was it added to metadata.finalizers when the CR was created?
The docs doesn't mention this situation. Could you give some references?

**Comment by estroz** [comment]
This problem is likely caused by the operator stopping before CRs are cleaned up. The finalizer is present so helm-operator can run `helm uninstall` prior to the CR being deleted. We might want to document this case in the tutorial with a note.

/kind documentation

**Comment by jberkhahn** [comment]
closed via #4461
